src/serialize.rs

```diff
@@ -0,0 +1,12 @@
+use serde::ser;
+
+pub fn hidden_or_not_set<T, S>(ref item: &Option<T>, serializer: S) -> Result<S::Ok, S::Error>
```

Comment:
...hmm, no I don't, the thing I was thinking of below won't work.

We could just keep it simple and use `#[serde(skip_serializing)]` on those instead.

Reply:
I guess to make this more generic we could just not specify that it's an `Option` and match `Some` and anything else to show `[hidden]`, while `None` will show `[not set]`. 

Also, I thought about using `#[serde(skip_serializing)]`, but then the user wouldn't know if the config was set or not and that might be useful information. With `skip_serializing` that config won't even show on the logs.

Reply:
> With `skip_serializing` that config won't even show on the logs.

Yep, but that's fine for stuff that isn't an `Option` because there is no "not set" state. The deserializer will fail before we can log the config if those aren't set.

Reply:
This function was written with `Option` in mind. I guess if we wanted to scrub sensitive data from non-`Option` types we would just have to use a different approach. Like you said, if a value that is not an `Option` is not set, the whole thing would fail anyways. 

For this particular PR, everything that I want to hide is inside an `Option`, so I think we should think about hiding other types when we get there. For the `hmac_key` we can just use `#[serde(skip_serializing)]` or we could write some serializing function for those cases, something that will always return `[hidden]`, like:

```
pub fn hidden<T, S>(ref item: &T, serializer: S) -> Result<S::Ok, S::Error>
where
    S: ser::Serializer,
{
    serializer.serialize_str("[hidden]")
}
```

Reply:
Alright! I understood you wrong, yeah, haha. Glad we're on the same page though :)